### ISSUE TITLE
Add new encryption option for a variable

### ISSUE THREAD

**Issue by baptisteArno** [baptisteArno/typebot.io#627]

**Comment by klinux** [comment]
Hey @baptisteArno — we need this feature in production at my company, so I started an implementation and would love your feedback before investing in UI work.

**Draft PR:** #2458

**TL;DR design:** a new workspace-scoped `WorkspaceSecret` entity (separate from `Variable`), encrypted at rest with the same AES-256-GCM pattern as `Credentials`. Referenced from typebots via `{{$secrets.NAME}}` and resolved **only** inside explicitly marked server-side block executors (this first PR wires HTTP request block: URL, headers, body, basic auth). Secrets are never written to `Result.variables`, `SetVariableHistoryItem`, chat replies, or any client-visible payload — they're fetched, substituted, and discarded just-in-time.

**Why a separate entity instead of an `isSecret` flag on `Variable`:** `Variable` flows through ~35 call sites including text bubbles, client responses, code blocks executed in the browser, etc. A runtime flag is fragile — one missed check leaks the secret. A distinct type makes mixing a compile error.

I split the work into 3 PRs to keep reviews small: (1) schema + backend + HTTP block integration [this one, drafted], (2) builder UI (Workspace Settings → Secrets + autocomplete + save-time validation), (3) docs. **I'm holding #2 and #3 until you validate the design in #2458** so we don't waste effort if you want a different approach.

Main questions I'd love your read on, summarized in the PR:

1. Separate entity vs. `isSecret` flag on `Variable` — I went separate for safety, but happy to pivot.
2. `{{$secrets.NAME}}` syntax — alternatives welcome.
3. Should I extend server-side resolution to email/Stripe/forge blocks in this PR, or keep it HTTP-only and follow up?
4. Want it gated behind a feature flag so it can merge incrementally?
5. Migration file — generate myself or leave for you?

Happy to rework anything. Thanks for building this project! 🙏

**Comment by klinux** [comment]
Hey @baptisteArno — following up since #2458 was closed as stale without discussion.

No worries on the timing, but before I invest more time I'd like to know whether this feature is something you'd want in typebot at all. A quick signal works:

- 👍 **interested** — I'll rework based on whatever direction you prefer (separate entity, flag on `Variable`, different syntax, smaller scope, etc.) and reopen.
- 👎 **not a priority / out of scope** — totally fine, feel free to close this issue and I'll maintain it in a fork.

Either answer is helpful. Thanks!

**Comment by baptisteArno** [comment]
Hey Kleber,

Thanks for following up, and sorry for closing #2458 as stale without a
clearer answer.

After thinking about it, I don't think this is something I want to support
in Typebot core right now. It touches sensitive boundaries around
variables, logs, results, exports, the public runtime, and self-hosted
deployments. I don't want to add it unless I can fully commit to
maintaining the security model around it.

For your company use case, the best path is to maintain this in your own
fork of Typebot. That will let you shape the secrets behavior around your
internal requirements without waiting on a core design.

Thanks again for taking the time to explore it.

Baptiste.

**Comment by baptisteArno** [comment]
Typebot is now in maintenance mode, so I’m no longer planning new features, product improvements or large internal projects. I’m closing this to keep the issue tracker focused on bugs, security and compatibility. Thanks for the suggestion!
